himbaechel/uarch/gatemate/pack.cc

```diff
@@ -283,6 +115,272 @@ void GateMatePacker::copy_constraint(NetInfo *in_net, NetInfo *out_net)
     }
 }
 
+void GateMatePacker::move_connections(NetInfo *from_net, NetInfo *to_net)
+{
+    for (const auto &usr : from_net->users) {
+        IdString port = usr.port;
+        usr.cell->disconnectPort(port);
+        usr.cell->connectPort(port, to_net);
+    }
+}
+
+void GateMatePacker::count_cell(CellInfo &ci)
+{
+    packed_cells.insert(ci.name);
+    if (!count_per_type.count(ci.type))
+        count_per_type[ci.type] = 1;
```

Comment:
I don't think there's any need for this whole conditional - `count_per_type[ci.type]` will create an entry with default (0) value in the dictionary if `ci.type` isn't yet in there, so just `count_per_type[ci.type]++;` unconditionally will always work